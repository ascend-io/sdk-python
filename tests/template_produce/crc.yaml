id: crc
component:
  readConnector:
    container:
      byteFunction:
        container:
          executable:
            code:
              source:
                inline: aW1wb3J0IGNyeXB0CmltcG9ydCByZXF1ZXN0cwoKVVJMID0gJ2h0dHBzOi8vcmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbS9DU1NFR0lTYW5kRGF0YS9DT1ZJRC0xOS9tYXN0ZXIvY3NzZV9jb3ZpZF8xOV9kYXRhL2Nzc2VfY292aWRfMTlfdGltZV9zZXJpZXMvdGltZV9zZXJpZXNfMTktY292aWQtRGVhdGhzLmNzdicKCgpkZWYgY29udGV4dChjcmVkZW50aWFsczogc3RyKToKICAgIHJldHVybiB7fQoKCmRlZiBsaXN0X29iamVjdHMoY29udGV4dDogZGljdCwgbWV0YWRhdGE6IGRpY3QpOgogICAgcmVzcCA9IHJlcXVlc3RzLmdldChVUkwpCiAgICB5aWVsZCB7J25hbWUnOiAnY29uZmlybWVkX2Nhc2VzJywgJ2ZpbmdlcnByaW50JzogY3J5cHQuY3J5cHQoc3RyKHJlc3AuY29udGVudCkpLCAnaXNfcHJlZml4JzogRmFsc2V9CgoKZGVmIHJlYWRfYnl0ZXMoY29udGV4dDogZGljdCwgbWV0YWRhdGE6IGRpY3QpOgogICAgcmVzcCA9IHJlcXVlc3RzLmdldChVUkwpCiAgICB5aWVsZCByZXNwLmNvbnRlbnQK
    bytes:
      parser:
        lambdaParser:
          code:
            inline: IyBpbXBvcnQgcGFja2FnZSBuZWVkZWQKIyBQbGVhc2UgYWRkIHRoZW0gdG8gaW5zdGFsbCBsaXN0IGJlbG93LAojIGlmIHRoZXkgYXJlIG5vdCBwcmUtYnVpbHQKaW1wb3J0IGNzdgppbXBvcnQgZGF0ZXRpbWUKaW1wb3J0IHRpbWUKCiMgbWFpbiBmdW5jdGlvbgpkZWYgcGFyc2VyX2Z1bmN0aW9uKHJlYWRlciwgb25OZXh0KToKICAgICMgVXNlIHJlYWRlciB0byByZWFkIGluIGRhdGEsIGFuZCBwYXNzIGEgcHl0aG9uIG1hcCB0byBvbk5leHQgZnVuY3Rpb24gYXMgcGFyc2luZyByZXN1bHQuCiAgICBjc3ZfcmVhZGVyID0gY3N2LnJlYWRlcigoYi5kZWNvZGUoJ3V0Zi04JykgZm9yIGIgaW4gcmVhZGVyKSkKICAgIGhlYWRlciA9IG5leHQoY3N2X3JlYWRlcikKICAgIHRpbWVzID0gW3RpbWUuc3RycHRpbWUoZCwgJyVtLyVkLyV5JykgZm9yIGQgaW4gaGVhZGVyWzQ6XV0KICAgIGRhdGVzID0gW2RhdGV0aW1lLmRhdGUoeWVhcj10LnRtX3llYXIsIG1vbnRoPXQudG1fbW9uLCBkYXk9dC50bV9tZGF5KS5pc29mb3JtYXQoKSBmb3IgdCBpbiB0aW1lc10KICAgIGZvciBsaW5lIGluIGNzdl9yZWFkZXI6CiAgICAgICAgY291bnRyeSA9IGxpbmVbMV0KICAgICAgICBzdGF0ZSA9IGxpbmVbMF0KICAgICAgICBmb3IgaSwgY291bnQgaW4gZW51bWVyYXRlKGxpbmVbNDpdKToKICAgICAgICAgICAgdHJ5OgogICAgICAgICAgICAgICAgb25OZXh0KHsnQ291bnRyeV9SZWdpb24nOiBjb3VudHJ5LCAnUHJvdmluY2VfU3RhdGUnOiBzdGF0ZSwgJ0RhdGUnOiBkYXRlc1tpXSwgJ0RlYXRocyc6IGNvdW50fSkKICAgICAgICAgICAgZXhjZXB0OgogICAgICAgICAgICAgICAgcmFpc2UgRXhjZXB0aW9uKGYnZXJyb3IgZm9yIHtpfToge2xpbmV9JykK
    updatePeriodical:
      offset: 103366736s
      period: 134217727s
