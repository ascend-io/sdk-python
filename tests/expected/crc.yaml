# id: crc
{%- set ascend_byte_function = ascend_include_raw(resourceParentDirectory, "crc_byte_function.py") %}
{%- set ascend_lambda_parser = ascend_include_raw(resourceParentDirectory, "crc_lambda_parser.py") %}
component:
  readConnector:
    bytes:
      parser:
        lambdaParser:
          code:
            inline: {{ ascend_lambda_parser | b64encode }}
    container:
      byteFunction:
        container:
          executable:
            code:
              source:
                inline: {{ ascend_byte_function | b64encode }}
    updatePeriodical:
      offset: 103366736s
      period: 134217727s
